syncer/syncer_test.go

```diff
@@ -272,20 +243,6 @@ func TestSynchronize_FailedInitialATXsSync(t *testing.T) {
 	require.False(t, ts.syncer.ListenToATXGossip())
 	require.False(t, ts.syncer.ListenToGossip())
 	require.False(t, ts.syncer.IsSynced(context.TODO()))
-
-	syncedCh := ts.syncer.RegisterForSynced(context.TODO())
-	select {
-	case <-syncedCh:
-		require.Fail(t, "node should not be synced")
-	case <-time.After(100 * time.Millisecond):
-	}
-
-	atxSyncedCh := ts.syncer.RegisterForATXSynced()
-	select {
-	case <-atxSyncedCh:
-		require.Fail(t, "node should not be atx synced")
-	case <-time.After(100 * time.Millisecond):
```

Comment:
Same here. If ATX Sync failed, `RegisterForATXSynced` should return a blocking channel.

Reply:
why isn't require.False(t, ts.syncer.ListenToGossip()) sufficient?

Reply:
Because after that change we have no test that checks that `RegisterForATXSynced` returns a blocked channel when the state is "not synced". Right now it does when `ts.syncer.ListenToGossip()` returns `false`, but implementations change over time.

The question is should we test as if `syncer` is a black box or not? The former leads inevitably to more tests and more test coverage, the later requires someone changing the implementation to what and how it is used in other places so they don't break things.